textalyze.js

```diff
@@ -39,6 +39,31 @@ function itemCounts(array) {
   }, new Map());
 }
 
+/**
+ * Given a count of chars in a text, gets its map of chars and occurences, and calculates a map of chars and frequencies based
+ * on the passed count of chars.
+ * @param {Integer} totalCount - The total amount of chars in a text.
+ * @param {Map} itemCounts - A map containing the occurrences of each char in the text.
+ * @returns {Map} itemFrequencies - A map containing the frequencies of each char based on the total count.
+ */
+function itemFrequencies(totalCount, itemCounts) {
+  if (totalCount <= 0) {
+    throw new Error('The items frequencies can\'t be computed without a total count of chars.');
```

Comment:
Throw the most specific error that makes sense.  In this case, [RangeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RangeError) is what you want.

That said, you don't really need to pass in `totalCount` since you can calculate it by summing `itemCounts.values()`:

```javascript
const totalCount = Array.from(itemCounts.values()).reduce((x, y) => x + y);
```

Reply:
Much more expressive than using anonymous functions. This syntax is very similar to Swift closures.

Reply:
Thanks for showing me that.